app/core/scheduler/modes.py

```diff
@@ -1,78 +1,127 @@
-        ...
+    pass
 
 
 class ValidationMode(SchedulerMode):
-    def schedule(self, start: Time, end: Time):
-        signal.signal(signal.SIGINT, signal.SIG_IGN)
-
-        programs = read_ocs_zipfile(os.path.join(ROOT_DIR, 'app', 'data', '2018B_program_samples.zip'))
-
-        # Create the Collector and load the programs.
-        print('Loading programs...')
-        collector = Collector(
-            start_time=start,
-            end_time=end,
-            time_slot_length=config_collector.time_slot_length,
-            sites=config_collector.sites,
-            semesters=config_collector.semesters,
-            program_types=config_collector.program_types,
-            obs_classes=config_collector.obs_classes
-        )
-        collector.load_programs(program_provider=OcsProgramProvider(),
-                                data=programs)
-
-        selector = Selector(collector=collector)
-
-        # Execute the Selector.
-        # Not sure the best way to display the output.
-        selection = selector.select()
-        # Execute the Optimizer.
-        dummy = DummyOptimizer()
-        optimizer = Optimizer(selection, algorithm=dummy)
-        plans = optimizer.schedule()
-
-        # Save to database
-        PlanManager.set_plans(plans)
+    """Validation mode is used for validate the proper functioning
+
+    Attributes:
+        _obs_statuses_to_ready (ClassVar[FrozenSet[ObservationStatus]]): 
+            A set of statuses that show the observation is Ready.
+    """
+
+    # The default observations to set to READY in Validation mode.
+    _obs_statuses_to_ready: ClassVar[FrozenSet[ObservationStatus]] = (
+        frozenset([ObservationStatus.ONGOING, ObservationStatus.OBSERVED])
+    )
+
+    @staticmethod
+    def _clear_observation_info(obs: Iterable[Observation],
+                                obs_statuses_to_ready: FrozenSet[ObservationStatus],
+                                observation_filter: Optional[Callable[[Observation], bool]] = None) -> NoReturn:
+        """
+        Given a single observation, clear the information associated with the observation.
+        This is done when the Scheduler is run in Validation mode in order to start with a fresh observation.
+
+        This consists of:
+        1. Setting an observation status that is in obs_statuses_to_ready to READY (default: ONGOING or OBSERVED).
+        2. Setting used times to 0 for the observation.
+
+        Additional filtering may be done by specifying an optional filter for observations.
+        """
+        if observation_filter is not None and not observation_filter(obs):
+            return
+
+        for o in obs:
+            for atom in o.sequence:
+                atom.prog_time = timedelta()
+                atom.part_time = timedelta()
+
+            if o.status in obs_statuses_to_ready:
+                o.status = ObservationStatus.READY
+
+    @staticmethod
+    def build_collector_with_clear_info(func: Callable):
```

Comment:
Every change is done with a decorator. So each mode now store different decorators that modify the `builder_` method of the specific component. Right now we just have one action ( clear obs info) but in the future we are probably going to replace those with `build_validation_collector` or something like that.